cryptography/src/bls12381/scheme.rs

```diff
@@ -98,3 +98,502 @@ impl Scheme for Bls12381 {
         (group::G1_ELEMENT_BYTE_LENGTH, group::G2_ELEMENT_BYTE_LENGTH)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    /// Tests use the bls12381 test vectors
+    /// from https://github.com/ethereum/bls12-381-tests (v0.1.2).
+    use super::{Bls12381, Scheme};
+    use crate::PrivateKey;
+    use crate::PublicKey;
+    use bytes::Bytes;
+
+    #[test]
+    fn sign() {
+        let tests_cases = [
+            vector_sign_1(),
+            vector_sign_2(),
+            vector_sign_3(),
+            vector_sign_4(),
+            vector_sign_5(),
+            vector_sign_6(),
+            vector_sign_7(),
+            vector_sign_8(),
+            vector_sign_9(),
+        ];
+
+        for (index, test_case) in tests_cases.iter().enumerate() {
+            test_vector_sign(
+                index + 1,
+                test_case.0.clone(),
+                test_case.1.clone(),
+                test_case.2.clone(),
+            );
+        }
+    }
+
+    #[test]
+    fn sign_zero_private_key() {
+        let v = vector_sign_10();
+        let private_key = PrivateKey::from(v.0);
+        let signer = <Bls12381 as Scheme>::from(private_key);
+        assert!(signer.is_none())
+    }
+
+    #[test]
+    fn verify() {
+        let tests_cases = [
+            vector_verify_1(),
+            vector_verify_2(),
+            vector_verify_3(),
+            vector_verify_4(),
+            vector_verify_5(),
+            vector_verify_6(),
+            vector_verify_7(),
+            vector_verify_8(),
+            vector_verify_9(),
+            vector_verify_10(),
+            vector_verify_11(),
+            vector_verify_12(),
+            vector_verify_13(),
+            vector_verify_14(),
+            vector_verify_15(),
+            vector_verify_16(),
+            vector_verify_17(),
+            vector_verify_18(),
+            vector_verify_19(),
+            vector_verify_20(),
+            vector_verify_21(),
+            vector_verify_22(),
+            vector_verify_23(),
+            vector_verify_24(),
+            vector_verify_25(),
+            vector_verify_26(),
+            vector_verify_27(),
+            vector_verify_28(),
+            vector_verify_29(),
+        ];
+
+        for (index, test_case) in tests_cases.iter().enumerate() {
+            test_vector_verify(
+                index + 1,
+                test_case.0.clone(),
+                test_case.1.clone(),
+                test_case.2.clone(),
+                test_case.3,
+            );
+        }
+    }
+
+    /// Runs the provided `sign` vector test.
```

Comment:
nit: Given you only use `test_vector_sign` and `test_vector_verify` in one place, just move the logic inline (in the two loops above)